app/classifier/tasks/survey/choice.jsx

```diff
@@ -0,0 +1,230 @@
+import React from 'react';
+import TriggeredModalForm from 'modal-form/triggered';
+import { Markdown } from 'markdownz';
+import Translate from 'react-translate-component';
+import ImageFlipper from './image-flipper';
+import Utility from './utility';
+
+
+class Choice extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleIdentification = this.handleIdentification.bind(this);
+    this.state = {
+      answers: Object.assign({}, props.annotationValue.answers),
+      focusedAnswer: ''
+    };
+  }
+
+  checkFilledIn() {
+    // if there are no questions, don't make them fill one in
+    if (Utility.getQuestionIDs(this.props.task, this.props.choiceID).length === 0) {
+      return true;
+    }
+
+    // if there are questions, it's fine as long as they've filled required ones in
+    const answerProvided = [];
+    Utility.getQuestionIDs(this.props.task, this.props.choiceID).map((questionId) => {
+      const question = this.props.task.questions[questionId];
+      if (question.required) {
+        const answer = this.state.answers[questionId];
+        if (answer && answer.length > 0) {
+          answerProvided.push(true);
+        } else {
+          answerProvided.push(false);
+        }
+      } else {
+        answerProvided.push(true);
+      }
+    });
+    return answerProvided.every(answer => (answer === true));
+  }
+
+  handleAnswer(questionId, answerId, e) {
+    const { answers } = this.state;
+    answers[questionId] = answers[questionId] ? answers[questionId] : [];
+    if (this.props.task.questions[questionId].multiple) {
+      if (e.target.checked) {
+        answers[questionId].push(answerId);
+      } else {
+        answers[questionId].splice(answers[questionId].indexOf(answerId), 1);
+      }
+    } else {
+      if (answerId === answers[questionId]) {
+        delete answers[questionId];
+        this.refs[questionId].checked = false;
```

Comment:
`this.refs` is deprecated.

Reply:
I'm not sure why this is setting the checked state of a checkbox directly. Surely, updating `state.answers` on the line above is enough to uncheck the question?

Maybe rewriting this should wait until we have tests for the survey task components.